### PR TITLE
Support `dst_N_autocreation` ZFS properties

### PR DESCRIPTION
**Pull request by jimklimov**
Builds on top of #636 ideas and is a step towards #503 hopefully.

At this time some optimization is possible: to only get dataset properties for the source tree once and walk the resulting array of hashes, instead of requesting props many times (once per child dataset as it does here, not unlike the `oracleMode` which trades one-by-one `zfs` commands handling vs. memory footprint of querying large chunks of names/properties on very populated ZFS trees). Still, as an MVP, this code already is functional (checked locally).

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -5,6 +5,7 @@ znapzend (0.21.3) unstable; urgency=medium
   * Extended handling of "org.znapzend:enabled=off" setting for sub-trees: now if the same intermediate dataset declares "org.znapzend:recursive=on", the disablement of znapzend handling takes place for descendants as well (previously it had effect only exactly for datasets that set "org.znapzend:enabled=off" as a local ZFS property)
   * Fixed CI recipes and contents for spell-checker
   * Added rc-script and integration documentation for FreeBSD and similar platforms
+  * Converted configure.ac and numerous Makefile.am to avoid GNU Make syntax in favor of portability: tested with Solaris/illumos Sun make and with FreeBSD make
```
Comment (oetiker) [inline CHANGES:8]: hmmm that seems not the match the changes

**Review by oetiker — APPROVED** [review]

**Comment by oetiker** [comment]
@jimklimov the last open pr ... could you have a look at the conflicts?

**Comment by jimklimov** [comment]
Cool! :) Yes, thanks. Were the others squashed too, I guess (should have otherwise shared part of history in commit chains IIRC)?

**Comment by jimklimov** [comment]
PR source updated to match master (apply original changes over it).

**Review by oetiker — APPROVED** [review]

**Comment by oetiker** [comment]
not squashed :)

**Pull request merged**